app/services/hyrax/visibility_intention.rb

```diff
@@ -66,13 +66,28 @@ def wants_embargo?
     ##
     # @return [Boolean]
     def valid_lease?
-      wants_lease? && release_date.present?
+      wants_lease? && release_date.present? && a_valid_date?(release_date)
     end
 
     ##
     # @return [Boolean]
     def wants_lease?
       visibility == LEASE_REQUEST
     end
+
+    private
+
+    ##
+    # @param date [Object]
+    # @return [Boolean]
+    # @note If we don't have a valid date, we really can't have a
+    # valid release_date
+    def a_valid_date?(date)
+      return true if date.is_a?(Date)
+      return true if date.is_a?(Time)
+      Date.parse(date)
+    rescue
```

Comment:
should this be more carefully targeted? what do we want to `rescue`?

if it should catch `StandardError`, our practice is to include that explicitly for clarity.

Reply:
I tried catching a more specific exception, and was getting a `NameError` and parse time.  I can give it another attempt.